build_tools/aws-sdk-code-generator/lib/aws-sdk-code-generator/plugin_list.rb

```diff
@@ -15,6 +15,17 @@ def each(&block)
 
     private
 
+    def compute_plugin_options(class_name, options)
+      plugin_options = const_get(class_name).options
+      if class_name == 'Aws::Plugins::EndpointDiscovery' &&
```

Comment:
I don't love having the logic for this here because its specific to the business logic of one plugin.  Ideally - this should be pushed into the plugin itself.  And one option would be to allow doc_default to be a block that takes the `options`.